### ISSUE TITLE
dictionary changed size during iteration

### ISSUE THREAD

**Issue by kputland** [dpkp/kafka-python#657]
version: 1.0.1

```
2016-04-20 20:29:39,837 - kafka.producer.sender - ERROR - Uncaught error in kafka producer I/O thread
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/kafka/producer/sender.py", line 57, in run
    self.run_once()
  File "/usr/lib/python2.7/site-packages/kafka/producer/sender.py", line 116, in run_once
    self.config['request_timeout_ms'], self._metadata)
  File "/usr/lib/python2.7/site-packages/kafka/producer/record_accumulator.py", line 251, in abort_expired_batches
    for tp, dq in six.iteritems(self._batches):
RuntimeError: dictionary changed size during iteration
```

**Comment by dpkp** [comment]
Pls test w 1.0.2 and/or master. I believe this has been fixed.

**Comment by dpkp** [comment]
pls reopen if this is not fixed in 1.0.2 / master
